src/components/Renderer/index.tsx

```diff
@@ -25,14 +25,13 @@ export const getValue = (
 	schema?: JSONSchema,
 	uiSchema?: UiSchema,
 ) => {
-	const calculatedValue = get(uiSchema, 'ui:value', value);
+	const calculatedValue = uiSchema?.['ui:value'] ?? value;
 	// Fall back to schema's default value if value is undefined
-	return calculatedValue !== undefined
-		? calculatedValue
-		: get(schema, 'default');
+	return calculatedValue ?? schema?.default;
 };
 
-const widgetWrapperUiOptionKeys = keys(WidgetWrapperUiOptions);
+// Convert these to lodash paths once here to avoid having to do it for every single render
+const widgetWrapperUiOptionKeyPaths = keys(WidgetWrapperUiOptions).map(toPath);
```

Comment:
This preparsing is the main win, the path conversion in general is pretty quick but when rendering a large list of items it gets called enough times to be noticeable, and also allocates arrays each time which will contribute to time spent in the gc